src/realm-settings/user-profile/attribute/AttributeGeneralSettings.tsx

```diff
@@ -0,0 +1,350 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-condition */
+/* eslint-disable prettier/prettier */
+import React, { useState } from "react";
+import {
+  Divider,
+  FormGroup,
+  Radio,
+  Select,
+  SelectOption,
+  SelectVariant,
+  Switch,
+  TextInput,
+} from "@patternfly/react-core";
+import { useTranslation } from "react-i18next";
+import { HelpItem } from "../../../components/help-enabler/HelpItem";
+import { Controller, useFormContext } from "react-hook-form";
+import { FormAccess } from "../../../components/form-access/FormAccess";
+import { useAdminClient, useFetch } from "../../../context/auth/AdminClient";
+import type ClientScopeRepresentation from "@keycloak/keycloak-admin-client/lib/defs/clientScopeRepresentation";
+import "../../realm-settings-section.css";
+
+const ENABLED_REQUIRED_WHEN = ["Always", "Scopes are requested"] as const;
+const REQUIRED_FOR = [
+	{ label: "Both users and admins", value: ["admin", "user"]}, { label: "Only users", value: "user" },{ label: "Only admins", value: "admin" }
+  ] as const;
+
+export const AttributeGeneralSettings = () => {
+  const { t } = useTranslation("realm-settings");
+  const adminClient = useAdminClient();
+  const form = useFormContext();
+  const [clientScopes, setClientScopes] =
+    useState<ClientScopeRepresentation[]>();
+  const [selectEnabledWhenOpen, setSelectEnabledWhenOpen] = useState(false);
+  const [selectRequiredForOpen, setSelectRequiredForOpen] = useState(false);
+  const [isAttributeGroupDropdownOpen, setIsAttributeGroupDropdownOpen] =
+    useState(false);
+  const [enabledWhenSelection, setEnabledWhenSelection] = useState("Always");
+  const [requiredWhenSelection, setRequiredWhenSelection] = useState("Always");
+
+  useFetch(
+    () => adminClient.clientScopes.find(),
+    (clientScopes) => {
+      setClientScopes(clientScopes);
+    },
+    []
+  );
+
+  return (
+    <FormAccess role="manage-realm" isHorizontal>
+      <FormGroup
+        label={t("attributeName")}
+        labelIcon={
+          <HelpItem
+            helpText="realm-settings-help:attributeNameHelp"
+            fieldLabelId="realm-settings:attributeName"
+          />
+        }
+        fieldId="kc-attribute-name"
+        isRequired
+      >
+        <TextInput
+          isRequired
+          type="text"
+          id="kc-attribute-name"
+          name="name"
+          defaultValue=""
+          ref={form.register({
+            required: {
+              value: true,
+              message: `${t("validateName")}`,
+            },
+          })}
+          data-testid="attribute-name"
+        />
+        {form.errors.name && (
+          <div className="error">{form.errors.name.message}</div>
+        )}
+      </FormGroup>
```

Comment:
You can simplify this a bit like this:
```suggestion
      <FormGroup
        label={t("attributeName")}
        labelIcon={
          <HelpItem
            helpText="realm-settings-help:attributeNameHelp"
            fieldLabelId="realm-settings:attributeName"
          />
        }
        fieldId="kc-attribute-name"
        isRequired
        helperTextInvalid={t("common:required")}
        validated={
          form.errors.name ? "error" : "default"
        }
      >
        <TextInput
          type="text"
          id="kc-attribute-name"
          name="name"
          defaultValue=""
          ref={register({ required: true })}
          data-testid="attribute-name"
          validated={
             form.errors.name ? "error" : "default"
           }
        />
      </FormGroup>
```